src/yunohost/tools.py

```diff
@@ -555,3 +555,63 @@ def tools_diagnosis(auth, private=False):
         diagnosis['private']['domains'] = domain_list(auth)['domains']
 
     return diagnosis
+
+
+def tools_urlavailable(auth, url):
+    """
+    Check availability of a web path
+
+    Keyword argument:
+        url -- Url to check
+    """
+
+    domain, path = _parse_app_url(url)
+
+    # Abort if domain is unknown
+    if domain not in domain_list(auth)['domains']:
+        raise MoulinetteError(errno.EINVAL, m18n.n('domain_unknown'))
+
+    # Fetch apps map
+    apps_map = app_map(raw=True)
+
+    # Loop through all apps to check if path is taken by one of them
+    available = "Yes"
+    if domain in apps_map:
+        # Loop through apps
+        for p, a in apps_map[domain].items():
+            if path == p:
+                available = "No"
+            # We also don't want conflicts with other apps starting with
+            # same name
+            elif path.startswith(p) or p.startswith(path):
+                available = "No"
+
+
+    return { "available" : available }
+
+def _parse_app_url(url):
+
+    # We want url to be of the format :
+    #  some.domain.tld/foo
+
+    # Remove http/https prefix if it's there
+    if url.startswith("https://"):
+        url = url[len("https://"):]
+    elif url.startswith("http://"):
+        url = url[len("http://"):]
+
+    # Remove trailing slash
+    url = url.rstrip("/")
+
+    # Check url contains at least a slash
+    if "/" not in url:
+        raise MoulinetteError(errno.EINVAL, m18n.n('invalid_url_format'))
+
+    # Now we extract the domain and uri/path
+    domain = url[:url.index('/')]
```

Comment:
I think that `url.split("/")[0]` would be nicer and also safer (since `string.index` will raise in the substring is not present).

Also this won't behave as expected if the domain is `domain.com/` will with `string.split` it will.